src/server/campaignd/server.cpp

```diff
@@ -515,40 +598,37 @@ void server::send_error(const std::string& msg, const std::string& extra_data, s
 	async_send_doc(sock, doc, std::bind(&server::handle_new_client, this, _1), null_handler);
 }
 
-void server::delete_campaign(const std::string& id)
+config& server::get_addon(const std::string& id)
 {
-	config::child_itors itors = campaigns().child_range("campaign");
-
-	std::size_t pos = 0;
-	bool found = false;
-	std::string fn;
-
-	for(config& cfg : itors) {
-		if(cfg["name"] == id) {
-			fn = cfg["filename"].str();
-			found = true;
-			break;
-		}
-
-		++pos;
+	auto addon = addons_.find(id);
+	if(addon != addons_.end()) {
+		return addons_.at(id);
+	} else {
+		return config::get_invalid();
```

Comment:
You just obtained a valid iterator to the item you were looking for and instead of dereferencing said iterator you're searching for the same item all over again.

Reply:
Thanks! Changed to addon->second .